src/Helpers/Post_Data_Helper.php

```
@@ -0,0 +1,170 @@
+	// phpcs:ignore Universal.Operators.StrictComparisons.LooseNotEqual -- Intentional loose comparison for type coercion
+		return $old_value != $new_value;
+	}
+
+	/**
+	 * Check if meta value has changed
+	 *
+	 * Combines value retrieval and comparison for meta fields.
+	 * Used by meta trackers to determine if a meta field update represents an actual change.
+	 *
+	 * @param int    $object_id  Post ID.
+	 * @param string $meta_key   Meta key being updated.
+	 * @param mixed  $new_value  New meta value.
+	 * @return bool True if the meta value has actually changed
+	 */
+	public static function has_meta_value_changed( int $object_id, string $meta_key, mixed $new_value ): bool {
+		// Always fetch current value from database for accurate comparison
+		$current_value = get_post_meta( $object_id, $meta_key, true );
+		return self::has_value_changed( $current_value, $new_value );
+	}
+
+	/**
+	 * Flag a field for sync across all translation posts
+	 *
+	 * Unified function for flagging both content and meta fields for sync.
+	 * Stores pending updates directly on each translation post instead of source post.
+	 * Only processes changes from original/source language posts to prevent recursive tracking.
+	 *
+	 * @param int    $source_post_id Post ID where change occurred (must be original post).
+	 * @param string $field_name     Field name that changed (e.g., 'title', 'content', 'field_123').
+	 * @param string $field_type     Field type: 'content' or 'meta'.
+	 * @return bool True if fields were flagged successfully
+	 */
+	public static function flag_field_for_sync( int $source_post_id, string $field_name, string $field_type ): bool {
+		if ( ! get_post( $source_post_id ) ) {
+			return false;
+		}
+
+		// Only flag fields from original/source language posts.
+		// Translation posts should never trigger sync flags for other translations.
+		if ( ! WPML_Post_Helper::is_original_post( $source_post_id ) ) {
+			return false;
```

should not return false but throw an exception since it is an invalid argument